crates/router/src/core/payouts.rs

```diff
@@ -2461,6 +2427,28 @@ pub async fn make_payout_data(
         payouts::PayoutRequest::PayoutRetrieveRequest(_) => None,
     };
 
+    let merchant_connector_account =
+        if payout_attempt.connector.is_some() && payout_attempt.merchant_connector_id.is_some() {
+            let connector_name = payout_attempt
+                .connector
+                .clone()
+                .get_required_value("connector_name")?;
```

Comment:
We wouldn't need to do this again, if you used map / match for destructuring Optional field 👀